src/components/timers/Stopwatch.tsx

```diff
@@ -1,3 +1,149 @@
-const Stopwatch = () => null;
+import Button from '../generic/Button';
+import DisplayTime from '../generic/DisplayTime';
+import Input from '../generic/Input';
+import Panel from '../generic/Panel';
+
+import { useState, useEffect, useRef } from 'react';
+
+const Stopwatch = () => {
+    const [inputMinutes, setInputMinutes] = useState(0);
+    const [inputSeconds, setInputSeconds] = useState(0);
+    const [totalMilliseconds, setTotalMilliseconds] = useState(0);
+    const [isRunning, setIsRunning] = useState(false);
+    const [isPaused, setIsPaused] = useState(false);
+    const [isCompleted, setIsCompleted] = useState(false);
+    const intervalRef = useRef<number | null>(null);
+
+    const targetMilliseconds = (inputMinutes * 60000) + (inputSeconds * 1000);
+
+    // Time functions
+    const getMinutes = () => Math.floor(totalMilliseconds / 60000);
+    const getSeconds = () => Math.floor((totalMilliseconds % 60000) / 1000);
+    const getHundredths = () => Math.floor((totalMilliseconds % 1000) / 10);
+
+    // Stopwatch function
+    const tick = () => {
+        setTotalMilliseconds((prevMilliseconds) => {
+            if (prevMilliseconds < targetMilliseconds) {
+                return prevMilliseconds + 10;
+            } else {
+                fastForwardTimer();
+                return prevMilliseconds;
+            }
+        });
+    };
+
+    // Start timer function
+    const startTimer = () => {
+        setTotalMilliseconds(0);
+        setIsRunning(true);
+        intervalRef.current = setInterval(tick, 10);
+    };
+
+    // Pause timer function
+    const pauseTimer = () => {
+        setIsPaused(true);
+        if (intervalRef.current) clearInterval(intervalRef.current);
+    };
+
+    // Resume timer function
+    const resumeTimer = () => {
+        setIsPaused(false);
+        intervalRef.current = setInterval(tick, 10);
+    };
+
+    // Fast forward timer function
+    const fastForwardTimer = () => {
+        setTotalMilliseconds(targetMilliseconds);
+        setIsCompleted(true);
+        if (intervalRef.current) clearInterval(intervalRef.current);
```

Comment:
try to manager your intervals in a single place so its easier to not forget to clear it